src/Dispatcher.php

```diff
@@ -28,6 +30,13 @@ public function __construct(Resolver $resolver)
     public function dispatch(object $message, array $options = []): void
     {
         $handler = $this->resolver->resolve($message);
-        $handler($message, new Context($this, $options));
+        $deferred = new MessageIterator();
+        $context = new Context($this, $deferred, $options);
+
+        $handler($message, $context);
+
+        foreach ($deferred as $deferredMessage) {
```

Comment:
как по мне, это неявно. а что если внутри контекста создавать свой message iterator и потом просто через геттер его отдать?
```suggestion
        $context = new Context($this, $options);

        $handler($message, $context);

        foreach ($context->getDeferred() as $deferredMessage) {
```

Reply:
Был такой изначальный вариант, сошлись с @zloyuser на текущем

Reply:
насчет BC-break будешь что-либо менят?

Reply:
Формальные параметры конструктора Context - это детали внутренней реализации commandbus т.к. Context не будет создаваться напрямую в проекте, поэтому оставлю как есть